core/blockchain/impl/block_storage_impl.cpp

```diff
@@ -197,16 +197,19 @@ namespace kagome::blockchain {
     }
     auto &existing_data = existing_block_data_opt.value();
 
+    auto move_if_flag = [](bool flag, auto&& value) {
+      if (flag) {
+        return std::move(value);
+      }
+      return std::optional<typename std::decay_t<decltype(value)>::value_type>(std::nullopt);
+    };
+
     // add all the fields from the new block_data
-    to_insert.header =
-        remove_flags.header ? std::nullopt : existing_data.header;
-    to_insert.body = remove_flags.body ? std::nullopt : existing_data.body;
-    to_insert.justification =
-        remove_flags.justification ? std::nullopt : existing_data.justification;
-    to_insert.message_queue =
-        remove_flags.message_queue ? std::nullopt : existing_data.message_queue;
-    to_insert.receipt =
-        remove_flags.receipt ? std::nullopt : existing_data.receipt;
+    to_insert.header = move_if_flag(remove_flags.header, std::move(existing_data.header));
```

Comment:
Looks unformatted. Use `clang-format on/of` comment to off formatting explicitly if it needed.